File: introcc_programacion/2ndEXAM/Main.cpp
#include <iostream>
using namespace std;

int sumDigits(int n){
    int sum = 0;
    while (n > 0){
        sum += n % 10;
        n /= 10;
    }
    return sum;
}

int primenumber(int n){
    if (n < 2 ){return 0;}
    for (int i = 2; i < n; i++){
        if ((n != 2) && (n % i == 0)){
            return 0;}
    }
    return 1;
}


int main(){

    int floor, ceil;
    cin >> floor;
    cin >> ceil;
    for (int i = floor; i <= ceil; i++){
        if (primenumber(i) == 1){
            int primeDigits = sumDigits(i);
            if (primenumber(primeDigits) == 1){
                cout << i << endl;
            }


        }
    }

    return 0;
}

File: introcc_programacion/2ndEXAM/Main.py
def printabundant(abundant):
    for i in abundant:
        print(i)

def isabundant(floor, ceil, abundant):
    for i in range(floor,ceil+1):
        sum=0
        for j in range(1,i+1):
            if i%j==0:
                sum+=j
        if sum>(2*i):
            abundant.append(i)

def getnumber():
    num = int(input())
    return num

def run():
    floor= getnumber()
    ceil= getnumber()
    abundant=[]
    isabundant(floor,ceil,abundant)
    printabundant(abundant)


if __name__ == "__main__":
    run()